### PR TITLE
40 Obsługa permisji

### PR DESCRIPTION
**Pull request by Mich519**
Obsługa permisji na backendzie:
- ustawianie permisji
- pobieranie permisji

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -15,76 +16,118 @@ def connect(sid, environ):
...
+
 @sio.event
-def send_video_event(sid, data):
-    print(sid, data)
-    sio.emit(event="receive_video_event", data=data['eventInfo'], room=data['roomId'], skip_sid=sid)
+def send_video_event(sid, eventInfo):
+    print(sid, eventInfo)
+    my_room_id = get_room_id_by_sid(sid, room_dict)
+    if my_room_id == "ROOM_NOT_FOUND":
+        return "ROOM_NOT_FOUND"
```
Comment (piotr-tomczyk) [inline src/main.py:48]: cannot we just return `my_room_id` here and in the other functions?

File: src/main.py [hunk 2]
```diff
@@ -15,76 +16,118 @@ def connect(sid, environ):
...
-    return room_dict[sid]['room']
+    
+    # prevent user from changing his own permissions
+    sid_server_permissions = room_dict[my_room_id][sid]['permissions']
+    sid_received_permissions = user_permissions[sid]['permissions']
+    if sid_server_permissions != sid_received_permissions:
+        sio.emit(event='permissions', data=room_dict[my_room_id], room=my_room_id)
+        return "OPERATION_NOT_ALLOWED"
+    
+    # TODO: prevent changing admin's permissions
```
Comment (piotr-tomczyk) [inline src/main.py:99]: Has it been implemented?

**Review by piotr-tomczyk — APPROVED** [review]
Add this TODO to the separate issue, and it would be good if you would fix this nitpicks but not obligatory

File: src/main.py (hunk 2 above)
Comment (Mich519) [reply src/main.py:99]: Not yet. I will move this to separate issue

**Comment by Mich519** [comment]
'kick' and 'chat' permissions are now validated in backend. 

- When user tries to kick someone and doesn't have 'kick' permission then "OPERATION_NOT_ALLOWED" response is returned.
- When user tries to send chat message and doesn't have 'chat' permission then "OPERATION_NOT_ALLOWED" response is returned.

**Review by juliasch5 — APPROVED** [review]

**Pull request merged**